lib/segment/sandbox.ex

```diff
@@ -3,20 +3,35 @@ defmodule Segment.Sandbox do
   Provides an api that can be used in tests to ensure
   data is correctly sent to segment.io.
   """
-
   alias Segment.{Track, Identify, Screen, Alias, Group, Page, Context}
   use Agent
 
+  @init_struct %{
+    track_calls: [],
+    identify_calls: [],
+    screen_calls: [],
+    alias_calls: [],
+    group_calls: [],
+    page_calls: []
+  }
+
   def start_link(_args) do
-    Agent.start_link(fn -> %{} end, name: __MODULE__)
+    Agent.start_link(fn -> @init_struct end, name: __MODULE__)
+  end
+
+  @doc """
+  Checkout function provides means to reset sandbox state to blank initial state.
+  """
+  def checkout() do
+    Agent.update(__MODULE__, fn _state -> @init_struct end)
   end
 
   def track(t = %Track{}) do
-    Agent.update(__MODULE__, &Map.put(&1, "track", t))
+    Agent.update(__MODULE__, fn state -> %{state | track_calls: [t | state.track_calls]} end)
   end
 
-  def get_track do
-    Agent.get(__MODULE__, &Map.get(&1, "track"))
+  def get_track_calls do
+    Agent.get(__MODULE__, &Map.get(&1, :track_calls))
```

Comment:
The list that is being returned should be reversed if we care about keeping the order of the events in which they arrived

Reply:
Should we return it that way ourselves or at least add some documentation to that effect?

Reply:
@deniskulicek ping on this again

Reply:
I can see use-cases that might require order to be preserved, though to keep it simple I'm fine with just making it clear that returned events aren't in order they arrived in.

They can be reversed quite easily when needed
`get_track_calls() |> Enum.reverse()`